runtime/common/src/asset_registry.rs

```diff
@@ -54,3 +56,20 @@ impl AssetProcessor<CurrencyId, AssetMetadata<Balance, CustomMetadata>> for Cust
 }
 
 pub type AssetAuthority = AsEnsureOriginWithArg<EnsureRoot<AccountId>>;
+
+
+pub struct FixedConversionRateProvider<OrmlAssetRegistry>(PhantomData<OrmlAssetRegistry>);
+
+impl<
+		OrmlAssetRegistry: Inspect<
+			AssetId = CurrencyId,
+			Balance = Balance,
+			CustomMetadata = asset_registry::CustomMetadata,
+		>,
+	> FixedConversionRateProviderTrait for FixedConversionRateProvider<OrmlAssetRegistry>
+{
+	fn get_fee_per_second(location: &MultiLocation) -> Option<u128> {
+		let metadata = OrmlAssetRegistry::metadata_by_location(location)?;
+		Some(metadata.additional.fee_per_second)
```

Comment:
It might be best if we try to come up with a default fee in case there is no fee stored for the asset in the registry. Something similar to [this](https://github.com/centrifuge/centrifuge-chain/blob/976d63612c94773359b1d8de16f2cd1aeb5a79f6/runtime/common/src/xcm.rs#L48).

Reply:
I saw that example but wasn't sure if it was best to define a default that may be insecure given a possible low token value, or simply do not allow to pay fee on that asset, which is what I decided in the end. Because I see that in the example they define a base fee based on the decimals, but this still makes no assumption of the possible value of that token.

Any idea how can we define a default that we can ensure is safe, even with low value assets?

Reply:
As discussed in the sync, I agree it's best to return with an error here and not implement a default `fee_per_second`. In practice, the `fee_per_second` should always be available for each asset we want to support.

Reply:
Allright! The error would eventually be returned [here](https://github.com/open-web3-stack/open-runtime-module-library/blob/91942bcc134c5f073b3198d2eb9ebdec48d8547f/asset-registry/src/impls.rs#L108), given that the fee would not be found, and it will just return `TooExpensive`. I do not know how a different error could be returned without modifying `AssetRegstiryTrader`, but if you think it is relevant we can implement it locally and modify that.

Reply:
Yes, sorry the `TooExpensive` error is enough. I was referring to that but phrased it poorly, so I'm fine if we keep this as is 👍